### ISSUE TITLE
Unknown modifier 'r'

### ISSUE THREAD

**Issue by Simbiat** [kalessil/phpinspectionsea#2010]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), 2026.2 |
| Language level | PHP 8.5                        |

Current behaviour
--
Have PHP 8.4+ and code like
```php
if (\preg_match('/something/iur', $string) === 1) {
    //Do something
}
```
`preg_match` gets highlighted because of unknown modifier `r`

Expected behaviour
--
No highlights, since `r` has been introduced in PHP 8.4. It may make sense to highlight it if either `u` or `i` (or both) is missing, though, since it will be redundant in that case.

Environment details
---

```
PhpStorm 2026.1.1
Build #PS-261.23567.149, built on April 23, 2026
Source revision: 1f6109ef9597a
Licensed to Dmitrii Kustov
You have a perpetual fallback license for this version.
Subscription is active until August 11, 2028.
Runtime version: 25.0.2+1-b329.117 amd64137.0.17-261-b81
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Toolkit: sun.awt.windows.WToolkit
Windows 11.0
Exception reporter ID: 1511241ccc5cd28-7089-4139-88ab-8aecba6eb3b2
GC: G1 Young Generation, G1 Concurrent GC, G1 Old Generation
Memory: 5120M
Cores: 16
Registry:
  debugger.new.tool.window.layout=true
  run.processes.with.pty=TRUE
  ide.experimental.ui=true
  llm.chat.session.feedback.widget.enabled=true
  llm.chat.default.agent.rollout.enabled=true
  trace.state.event.service.url=https://api.jetbrains.cloud/trace-status
  eslint.service.node.path=--max_old_space_size=12000
  terminal.new.ui=true
Non-Bundled Plugins:
  ru.adelf.idea.dotenv (261.23567.176)
  org.jetbrains.completion.full.line (261.23567.182)
  com.intellij.ml.llm (261.23567.193)
  Subversion (261.23567.176)
  com.github.xepozz.crontab (2026.1.2)
  com.intellij.mermaid (0.0.27+IJ.253)
  intellij.javascript.bun (261.22158.180)
  com.intellij.mcpServer (261.23567.174)
  com.kalessil.phpStorm.phpInspectionsUltimate-mp (2026.2)
  de.espend.idea.php.annotation (12.1.0)
  org.nette.neon (1.1.0)
  fr.adrienbrault.idea.symfony2plugin (2026.1.303)
  com.laravel_idea.plugin (13.1.2.261)
```

**Comment by Simbiat** [comment]
Interestingly, it's not highlighted in
```php
public const string ALLOWED_IN_BODY = '/^\s*(alternate )?((^| )(dns-prefetch|modulepreload|pingback|preconnect|prefetch|preload|stylesheet)( |$))+(alternate)?\s*$/uir';
```

**Comment by ea-inspections-team** [comment]
Thank you for reporting @Simbiat!
The false positive has been resolved.
